robocorp-code/vscode-client/src/extension.ts

```diff
@@ -324,6 +326,102 @@ async function cloudLogoutAndRefresh() {
+            },
+            async (
+                progress: vscode.Progress<{ message?: string; increment?: number }>,
+                token: vscode.CancellationToken
+            ): Promise<string | undefined> => await download(url, progress, token, location)
+        );
+    }
+    return location;
+}
+
+async function convertProject() {
+    const DEFAULT_ERROR_MSG = `
+            Could not convert project.
+            Please check the output logs for more details.
+            `;
+
+    const DEFAULT_ERROR_STATUS = "Error while converting project.";
+
+    const converterLocation = await ensureConvertBundle();
+    try {
+        const converterBundle = require(converterLocation);
+        window.setStatusBarMessage("Project conversion has started...");
+
+        // let the user decide what type of project will be converted
+        const vendorMap = {
+            "Blue Prism": "blueprism",
+            "UiPath": "uipath",
+            "Automation Anywhere 360": "a360",
+        };
+        const items = Object.keys(vendorMap);
+        let selectedItem = await window.showQuickPick(items, {
+            "placeHolder": "Please select vendor of current project",
+            "canPickMany": false,
+            "ignoreFocusOut": false,
+        });
+        window.setStatusBarMessage(`Selected project type: ${selectedItem}`);
+        if (!selectedItem) {
+            window.setStatusBarMessage("Project conversion was canceled.");
+            return;
+        }
+
+        // let the user decide where the conversion result will be saved
+        const destination = await window.showOpenDialog({
+            title: "Select conversion destination",
+            openLabel: "Select Output Folder",
+            canSelectFiles: false,
+            canSelectFolders: true,
+            canSelectMany: false,
+        });
+        if (!destination || destination.length == 0) {
+            window.setStatusBarMessage("Project conversion was canceled.");
+            return;
+        }
+
+        window.setStatusBarMessage("Project conversion is starting...");
+        const activeTextEditor = window.activeTextEditor;
+        const conversionResult = await converterBundle.convert(
+            vendorMap[selectedItem],
+            activeTextEditor.document.getText()
+        );
+        window.setStatusBarMessage("Done converting project.");
+        if (!converterBundle.isSuccessful(conversionResult)) {
+            logError("Error while converting project", new Error(conversionResult.error), "EXT_CONVERT_PROJECT");
+            window.setStatusBarMessage(DEFAULT_ERROR_STATUS);
+            window.showErrorMessage(DEFAULT_ERROR_MSG);
+            OUTPUT_CHANNEL.show();
+            return;
+        }
+        const populate_action: { result: string; success: boolean; message: string | null } | null =
+            await commands.executeCommand("robocorp.convertProject.internal", {
+                "destinationFolderURI": destination[0].path,
```

Comment:
The var names says it's an `uri`, yet you're passing a `path`... this doesn't seem right.

In general you either pass the uri directly (just do a `toString()` with it) and then in the python side you need to do `robocorp_ls_core.uris.to_fs_path` to access the filesystem path in the language server or you pass a path -- i.e.: `destinationFolderPath` and pass the `uri.fsPath`.

Passing the `uri.toString()` is preferred though as most of the language server APIs deal with `uris` and not `paths` (it's converted only when needed).